rego/rego.go

```diff
@@ -1416,6 +1433,36 @@ func (r *Rego) PrepareForEval(ctx context.Context, opts ...PrepareOption) (Prepa
 			},
 		},
 	})
+
+	if r.target == targetWasm {
+
+		if r.hasWasmModule() {
+			return PreparedEvalQuery{}, fmt.Errorf("wasm target not supported")
+		}
+
+		var modules []*ast.Module
+		for _, module := range r.compiler.Modules {
+			modules = append(modules, module)
+		}
+
+		queries := []ast.Body{r.compiledQueries[evalQueryType].query}
+
+		cr, err := r.compileWasm(modules, queries, evalQueryType)
+		if err != nil {
+			return PreparedEvalQuery{}, err
+		}
+
+		data, err := r.store.Read(ctx, r.txn, storage.Path{})
+		if err != nil {
+			return PreparedEvalQuery{}, err
+		}
+
+		r.opa, err = opa.New().WithPolicyBytes(cr.Bytes).WithDataJSON(data).Init()
```

Comment:
Is it possible that we're getting into a weird situation here? If `err != nil`, we'll set `r.opa = nil`. is that going to be a problem? (Alternatively, we'd do `o, err := opa.New()...; if err != nil { ... }; r.opa = o`)

Reply:
I don't think so but I'll update it nevertheless.